ImGuiFileDialog.cpp

```diff
@@ -75,300 +103,330 @@ ImGuiFileDialog::~ImGuiFileDialog()
 }
 
 /* Alphabetical sorting */
-static int alphaSort(
-	const void *a, const void *b)
+static int alphaSort(const struct dirent **a, const struct dirent **b)
 {
-	return strcoll(((dirent*)a)->d_name, ((dirent*)b)->d_name);
+    return strcoll((*a)->d_name, (*b)->d_name);
 }
 
-static bool stringComparator(FileInfoStruct a, FileInfoStruct b)
-{
-	bool res;
-	if (a.type != b.type) res = (a.type < b.type);
-	else res = (a.fileName < b.fileName);
-	return res;
-}
 
 void ImGuiFileDialog::ScanDir(std::string vPath)
 {
-	struct dirent **files = 0;
```

Comment:
I see people assign 0 to pointers all the time. I have to say I find it quite irritating. NULL should be used for pointers.

NULL has a value of zero, so in effect it is fine, but it helps the reader if you specify NULL. Even better, for C++, use nullptr as it helps the compiler too.